src/components/common/form/Form.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ActivityIndicator, StyleSheet, View } from "react-native";
+import { useForm } from "react-hook-form";
+import FormInput from "./FormInput";
+import { COLOR_PALETTE } from "../../../utils/constants";
+import Button, { BUTTON_TYPES } from "../Button";
+
+const Form = ({ inputs, defaultValues, onSubmit, isLoading, buttonText }) => {
+	const { control, handleSubmit } = useForm({
+		defaultValues: defaultValues
+	});
+
+	const renderFormButton = () =>
+		isLoading ? (
```

Comment:
move isLoading state to the button, that way you can just render the button with a prop called isLoading